Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pickable.py

```diff
@@ -51,17 +52,36 @@ def parser(cls, parser, loggers=None):
         )
 
     @classmethod
-    def pickable(cls, commits):
-        filtered_commits = []
+    def pickable(cls, commits, repository, commits_story=None):
+        filtered_in = set()
+        all_commits = dict()
+
+        commits_story = commits_story or CommitsStory()
+
         for commit in commits:
+            commits_story.add(commit)
             title = commit.message.splitlines()[0]
+            all_commits[str(commit)] = commit
             if search('Cherry-pick', title) or search('Versioning', title):
                 continue
-            filtered_commits.append(commit)
+            filtered_in.add(str(commit))
 
         # FIXME: We can eliminate more classes of commits by reasoning about commit relationship
         #    For example, a revert of a non-pickable commit is not itself pickable
-        return filtered_commits
+        for ref in list(filtered_in):
+            commit = all_commits[ref]
+            relationships = Trace.relationships(commit, repository, commits_story=commits_story)
+            if not relationships:
+                continue
+            for rel in relationships:
+                if rel.type in Relationship.PAIRED and str(rel.commit) not in filtered_in:
+                    filtered_in.remove(ref)
+                    break
```

Comment:
What is the purpose of the `break` here and `:82`?

Reply:
No reason to keep looking through the relationships of a commit if we've already determined the commit should be excluded.